src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -222,385 +204,59 @@ Plugin::Plugin()
 
     OV_ITT_TASK_CHAIN(PLUGIN, itt::domains::NPUPlugin, "Plugin::Plugin", "NPUBackends");
     _backends = std::make_shared<NPUBackends>(backendRegistry, _globalConfig);
-    OV_ITT_TASK_NEXT(PLUGIN, "registerOptions");
-    _backends->registerOptions(*_options);
+
+    /// Fetch CID version and create properties list
+    OV_ITT_TASK_NEXT(PLUGIN, "FetchCompilerVer");
+    compilerVersion cid_ver = {0, 0, 0};
+    auto device = _backends->getDevice();
+    if (device) {
+        cid_ver = device->getCompilerVersion();
```

Comment:
For IMD, seems throw exception here? Maybe return default value in IDevice instead of excpetion.